### PR TITLE
Add the target architecture field to the manifest.proto

### PR DESCRIPTION
**Pull request by musaprg**
Add the target architecture of envoy builds to the `manifest.proto`. These changes are related to getenvoy arm support https://github.com/tetratelabs/getenvoy/issues/109.

### CODE REVIEW

**Comment by musaprg** [comment]
This PR may need to wait for https://github.com/tetratelabs/getenvoy-package/pull/103.

**Review by lizan — APPROVED** [review]

File: api/manifest.proto [hunk 1]
```diff
@@ -65,6 +65,13 @@ message Build {
       LINUX_GLIBC = 3;
    }
 
+   enum Architecture {
```
Comment (codefromthecrypt) [inline api/manifest.proto:68]: maybe add a comment (and possibly an upstream envoy link) of the source of the architecture choices? Knowing what this is derived from can help in wondering what the future choices might be also, as for example some call "arm64" "aarch64" and also sometimes there are qualifiers. like arm64v8.. similar for x86 vs amd64

**Comment by mathetake** [comment]
well, the proto has already migrated to getenvoy's main repo (https://github.com/tetratelabs/getenvoy/blob/master/api/manifest.proto) so shall we re-open this there? (And @codefromthecrypt could you complete the migration by deleting proto here?)

**Comment by codefromthecrypt** [comment]
ah right there was a chicken egg... originally I was going to delete this here after we did a release which included the proto in the same repo, but yeah I suppose we've now encountered a reason to eagerly purge.. and I like deleting things so sgtm

**Comment by codefromthecrypt** [comment]
@musaprg I'm closing this here. Since I cost you time, let me know if you'd like me to recreate it on a shared branch on the getenvoy repo.

**Comment by musaprg** [comment]
@codefromthecrypt I'll recreate this to getenvoy main repo :+1:

**Pull request closed without merging**